client/src/app/editor/file-outputs/file-outputs.component.ts

```diff
@@ -57,30 +60,64 @@ export class FileOutputsComponent implements OnChanges, OnInit {
   ngOnInit() {
     const outputFileId = this.route.snapshot.queryParamMap.get('preview');
 
-    if (outputFileId) {
-      this.hasOutput
-        .pipe(
-          filter(hasOutput => hasOutput),
-          mergeMap(() => this.dataSource.connect()),
-          mergeMap(outputFiles => outputFiles),
-          filter(outputFile => outputFile.id === outputFileId),
-          filter(outputFile => isImage(outputFile.name)),
-          take(1)
+    const executionSelected$ = this.executionId$.pipe(
+      switchMap(executionId => this.labExecutionService.executionExists(executionId).pipe(catchError(() => of(false)))),
```

Comment:
This really needs to be fixed. It can't be that executionExists will give us a permission error. Still have to verify this locally

Reply:
Yep, maybe it's only on my machine so it would be great if you can verify this on your machine. I have created an issue (#839) for this already. When you confirm this issue we can address it in a separate PR.